3_summarize/src/annual_thermal_metrics.R

```diff
@@ -0,0 +1,256 @@
+
+calculate_annual_metrics <- function(target_name, site_ids, file_pattern) {
+  
+  purrr::map(site_ids, function(id) {
+    
+    data_ready <- read_feather(sprintf(file_pattern, id)) %>% 
+      select(site_id, date = DateTime, starts_with("temp_")) %>% 
+      pivot_longer(cols = starts_with("temp_"), names_to = "depth", values_to = "wtr") %>% 
+      mutate(depth = as.numeric(gsub("temp_", "", depth))) %>% 
+      mutate(year = as.numeric(format(date, "%Y")))
+    
+    annual_metrics <- data_ready %>% 
+      group_by(site_id, year) %>% 
+      summarize(
+        # winter_dur_0_4 = winter_dur_0_4(),
+        # coef_var_30_60 = coef_var_30_60(),
+        # coef_var_0_30 = coef_var_0_30(),
+        # stratification_onset_yday = stratification_onset_yday(),
+        # stratification_duration = stratification_duration(),
+        # sthermo_depth_mean = sthermo_depth_mean(),
+        
+        peak_temp = peak_temp(date, wtr, depth),
+        gdd_wtr_0c = gdd_wtr_0c(date, wtr),
+        gdd_wtr_5c = gdd_wtr_5c(date, wtr),
+        gdd_wtr_10c = gdd_wtr_10c(date, wtr),
+        
+        # bottom_temp_at_strat = bottom_temp_at_strat(),
+        # schmidt_daily_annual_sum = schmidt_daily_annual_sum(),
+        
+        mean_surf_jas = mean_surf_jas(date, wtr, depth),
```

Comment:
similar here, maybe this function is more generic and accepts `c("July","August","September")`